misc/python/materialize/output_consistency/data_generation/generator.py

```diff
@@ -0,0 +1,102 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from materialize.output_consistency.expressions.expression import (
+    Expression,
+    ExpressionWithNArgs,
+)
+from materialize.output_consistency.operations.operation import OperationWithNArgs
+from materialize.output_consistency.operations.operation_provider import OPERATION_TYPES
+from materialize.output_consistency.values.data_provider import DATA_TYPES
+from materialize.output_consistency.values.data_type import RawValue
+
+
+def create_expressions() -> list[ExpressionWithNArgs]:
+    expressions = []
+
+    for data_type in DATA_TYPES:
+
+        if len(data_type.raw_values) == 0:
+            continue
+
+        for operation_type in OPERATION_TYPES:
+            combinations = get_combinations(
+                data_type.raw_values,
+                offset_value=data_type.raw_values[0],
+                length=operation_type.op_count,
+                with_self=True,
+                with_earlier=not operation_type.commutative,
+            )
+
+            for combination in combinations:
+                if not is_valid_combination(operation_type, combination):
+                    continue
```

Comment:
Excluding these combinations for now. Planned to run them in a separate query expecting errors as outcome.